proposals/3392/dataset_partitioning.md

```diff
@@ -0,0 +1,301 @@
+                  "field": {
+                     "type": "String"
+                  },
+                  "value": {
+                     "type": "String"
+                  },
+                  "comparison": {
+                     "enum": [
+                        "EQUAL",
+                        "GREATER_THAN",
+                        "GREATER_EQUAL_THAN",
+                        "LESS_THAN",
+                        "LESS_EQUAL_THAN"
+                     ]
+                  }
+               }
+            }
+         ]
+      },
+      "FilterInputDatasetFacet": {
+         "allOf": [
+            {
+               "$ref": "https://...OpenLineage.json#/$defs/InputDatasetFacet"
+            },
+            {
+               "type": "object",
+               "properties": {
+                  "condition": {
+                     "type": "#/$defs/BaseFilterCondition"
+                  }
+               }
+            }
+         ]
+      }
+   },
+   "type": "object",
+   "properties": {
+      "filter": {
+         "$ref": "#/$defs/FilterInputDatasetFacet"
+      }
+   }
+}
+
+```
+
+
+
+An example filter facet for SQL `SELECT * FROM table WHERE col1 = 7 and col2 = 9` would be:
+
+```json
+{
+  "filter": {
+    "$id": "https://openlineage.io/spec/facets/1-0-0/BinaryFilterCondition.json",
+    "left": {
+      "field": "col1",
+      "value": "7",
+      "comparison": "EQUAL"
+    },
+    "right": {
+      "field": "col2",
+      "value": "9",
+      "comparison": "EQUAL"
+    },
+    "operator": "AND"
+  }
+}
+```
+    
+Please mind that this approach also allows providing a list of partitions, as well as extending it to add regex patterns on partitioning read.
+Output partitions dataset facet
+
+```json
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "$id": "https://openlineage.io/spec/facets/1-0-0/FilterInputDatasetFacet.json",
+    "$defs": {
+        "PartitionOutputDatasetFacet": {
+            "allOf": [
+                {
+                    "$ref": "#/$defs/OutputDatasetFacet"
+                },
+                {
+                    "type": "object",
+                    "updated": {
+                      "type": "array",
+                      "items": {
+                        "types": "object",
+                        "additionalProperties": true
+                      }
+                    }
+                }
+           ]
+        }
+    },
+    "type": "object",
+    "properties": {
+        "partitions": {
+          "$ref": "#/$defs/PartitionOutputDatasetFacet"
+        }
+    }
+}
+```
+
+An example of this would be a dataset partitioned by `business_date` and `country`. A job run affecting many countries for a single date would affect multiple partitions and this would be described with a facet:
+
+```json
+{
+  "partitions": {
+    "updated": [
+      {
+        "business_date": "2024-10-15",
+        "country": "PL"
+      },
+      {
+        "business_date": "2024-10-15",
```

Comment:
This describes only Hive-style partitioning, but other datasources may have another partitioning schemas. For example, Postgres allows `LIST`, `RANGE` and `HASH` partitioning. So instead of including value of partitioning columns (which can be arbitrary, and different for each row in the dataset), users may want to know just the partition name (in Postgres partition is just another table bound to main one) or a list of partition column constraints (e.g. `business_at BETWEEN '2024-01-01' AND '2025-01-01' for `RANGE` partitioning).

Reply:
Ok